version: '3.7'

networks: 
    covid_network:
        driver: bridge

services:
    postgres:
        image: postgres:12.2-alpine
        env_file: 
            - ./.env
        ports: 
            - 54321:5432
        logging:
            options:
                max-size: 10m
                max-file: "3"
        volumes: 
            - covid:/var/lib/postgresql/data/
        networks: 
            [covid_network]

    webserver:
        image: puckel/docker-airflow:latest
        restart: always
        depends_on:
            - postgres
        env_file: 
            - ./.env
        logging:
            options:
                max-size: 10m
                max-file: "3"
        volumes:
            - ./dags:/usr/local/airflow/dags
            # - ./plugins:/usr/local/airflow/plugins
        ports:
            - "8080:8080"
        command: webserver
        healthcheck:
            test: ["CMD-SHELL", "[ -f /usr/local/airflow/airflow-webserver.pid ]"]
            interval: 30s
            timeout: 30s
            retries: 3
        networks: 
            [covid_network]

volumes: 
    covid: